version: '3.8'

volumes: 
  dynamodb_data:

services:
  dynamodb-local:
    command: "-jar DynamoDBLocal.jar -sharedDb -dbPath /home/dynamodblocal/data"
    user: root
    image: "amazon/dynamodb-local:latest"
    container_name: dynamodb-local-z-index
    ports:
      - "8000:8000"
    volumes:
      - dynamodb_data:/home/dynamodblocal/data
    working_dir: /home/dynamodblocal
    networks:
      - backend
  dynamo-seeder:
    image: "python:3.13-rc-bookworm"
    working_dir: /home/api/seeder/
    volumes:
      - ./:/home/api/seeder
    entrypoint: ["/bin/bash"]
    command: -c "chmod +x ./dynamodb_seeder_setup.sh 
      && ./dynamodb_seeder_setup.sh 
      && chmod +x ./init_database.sh 
      && ./init_database.sh --forever"
    links:
      - dynamodb-local
    depends_on:
      - dynamodb-local
    environment:
      - AWS_ACCESS_KEY_ID=test
      - AWS_SECRET_ACCESS_KEY=test
      - REGION=us-east-1
      - DYNAMODB_ENDPOINT=http://dynamodb-local:8000
      - DROP_TABLE_Z_ORDER=${DROP_TABLE_Z_ORDER:-false}
      - DROP_TABLE_COMPOSITE_INDEX=${DROP_TABLE_COMPOSITE_INDEX:-false}
    networks:
      - backend



networks:
  backend:
    name: aws_backend
    driver: bridge
